### DISCUSSION TITLE
how do i use consult to live preview a given list of files + line number

### DISCUSSION THREAD

**Discussion by wn** [minad/consult#1035] [Q&A]
Given ("/home/john/proj/apple.hh:5" "/home/john/proj/apple.hh:12" "/home/john/proj/banana.hh:42" ....), i would like to see a popup with those files, and navigating to them shows a live preview of the file, and pressing enter navigates to the file.

Ideally, same file results will be grouped together, similar to consult-ripgrep.

I think it has something to do with consult--read but im not sure how to get it to work.

Thank you very much!

**Comment by minad** [comment]
(out of scope)

This question goes a bit too far. My suggestion is to take a look at the source of `consult-grep`, or the more basic `consult-recent-file`. It should be possible to reuse most of that.
